instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraTableNameExtractor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.cassandra.v4_0;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * this class is separate to enable testing. when the {@link #extractTableNameFromQuery} method was
+ * embedded there was an issue with visibility, even if the method was public.
+ */
+public class CassandraTableNameExtractor {
+
+  private CassandraTableNameExtractor() {}
+
+  private static final Pattern tableNameRegex =
+      Pattern.compile(
+          ".*(?:FROM|INTO|UPDATE|TRUNCATE|(?:CREATE|ALTER|DROP) TABLE)\\s+(?:IF (?:NOT )?EXISTS\\s+)?([A-Z1-9_]+\\.([A-Z1-9_]+)|([A-Z1-9_]+))",
+          Pattern.CASE_INSENSITIVE);
```

Comment:
from https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md

> It is not recommended to attempt any client-side parsing of `db.statement` just to get this property

We also parse jdbc SQL for the table name, though, violating this, so ??? <-- @mateuszrzeszutek @iNikem @johnbley @anuraaga

If we are going to parse/capture the table name, it would be great to use it in the span name: `<db.operation> <db.name>.<db.sql.table>`

Reply:
we could extract the table name without parsing anything if the statement is a prepared statement, but if the statement is a simplestatement that's the only way I could find to get it out.

Reply:
We already parse the cassandra query to sanitize it - once #2113 is in operation & table name will be available for free (although the current sanitizer implementation cannot extract table name from create/drop table statements, but I suspect that those are rare cases anyway).

Reply:
Okay, I'll have a look at refactoring and basing the table extraction on this.

Reply:
done